packages/graph/hash_graph/lib/graph/src/datastore/mod.rs

```diff
@@ -152,16 +152,24 @@ impl fmt::Display for DatabaseConnectionInfo {
     }
 }
 
-/// Describes the API of a Datastore implementation
+/// Describes the API of a Datastore implementation.
+///
+/// # Errors
+///
+/// In addition to the errors described in the methods of this trait, each implementation may return
+/// additional, implementation-specific errors.
```

Comment:
Does this seem better?

> Implementations of this trait should ensure to add the checks outlined in the "Error" sections of the method docs. Further errors might also be raised depending on the implementation, e.g. connection issues.

Reply:
Or just

> In addition to the errors described in the methods of this trait, further errors might also be raised depending on the implementation, e.g. connection issues.

Reply:
Went with the second phrasing: 2440924e47af6b0d6d662723f7031892eabcbf9a